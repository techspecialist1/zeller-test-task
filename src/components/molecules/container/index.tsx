import React, {FC} from 'react';
import {SafeAreaView, View} from 'react-native';

import {styles} from './styles';

type ContainerProps = {
  children: React.ReactNode;
};

const Container: FC<ContainerProps> = ({children}) => {
  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>{children}</View>
    </SafeAreaView>
  );
};

export default Container;
