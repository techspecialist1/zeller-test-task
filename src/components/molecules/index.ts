import ListView from './listView';
import Container from './container';
import RadioButton from '../atoms/radioButton';
import SearchInput from '../atoms/searchInput';
import ActivityIndicator from '../atoms/activityIndicator';
import ErrorBox from '../atoms/errorBox';

export {
  ListView,
  ActivityIndicator,
  RadioButton,
  SearchInput,
  Container,
  ErrorBox,
};
