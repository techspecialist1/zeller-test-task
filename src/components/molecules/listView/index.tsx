import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

import {styles} from './style';

type ListViewProps = {
  item: {
    name: string;
    role: string;
  };
  onPress: () => void;
};

const ListView: React.FC<ListViewProps> = props => {
  const {item, onPress} = props;

  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.userList}>
        <View style={styles.profile}>
          <Text style={styles.profileText}>{item?.name?.slice(0, 1)}</Text>
        </View>
        <View style={styles.listName}>
          <Text style={styles.name}>{item?.name}</Text>
          <Text style={styles.role}>{item?.role}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ListView;
