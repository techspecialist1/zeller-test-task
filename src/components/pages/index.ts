import HomePage from './homePage';
import DetailPage from './detailPage';

export {HomePage, DetailPage};
