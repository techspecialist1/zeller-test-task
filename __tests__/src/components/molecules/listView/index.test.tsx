import React from 'react';
import renderer from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import {ListView} from '../../../../../src/components/molecules';

describe('ListView', () => {
  const mockOnPress = jest.fn();

  const defaultProps = {
    item: {
      name: 'Pankaj',
      role: 'Developer',
    },
    onPress: mockOnPress,
  };

  test('renders correctly with default props', () => {
    const component = renderer.create(<ListView {...defaultProps} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('calls onPress function when pressed', () => {
    const component = renderer.create(<ListView {...defaultProps} />);
    const instance = component.root;
    const touchableOpacity = instance.findByType(TouchableOpacity);

    touchableOpacity.props.onPress();

    expect(mockOnPress).toHaveBeenCalled();
  });
});
