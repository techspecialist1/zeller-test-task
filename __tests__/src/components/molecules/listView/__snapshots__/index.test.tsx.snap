// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 18.46153846153846,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E8F2FB",
          "borderRadius": 4.923076923076923,
          "height": 49.230769230769226,
          "justifyContent": "center",
          "width": 49.230769230769226,
        }
      }
    >
      <Text
        style={
          {
            "color": "#0171CD",
            "fontSize": 18.46153846153846,
            "textAlign": "center",
          }
        }
      >
        P
      </Text>
    </View>
    <View
      style={
        {
          "paddingLeft": 14.76923076923077,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 19.692307692307693,
          }
        }
      >
        Pankaj
      </Text>
      <Text
        style={
          {
            "color": "#000",
            "textTransform": "capitalize",
          }
        }
      >
        Developer
      </Text>
    </View>
  </View>
</View>
`;
